rskj-core/src/main/java/org/ethereum/config/SystemProperties.java

```diff
@@ -583,42 +587,54 @@ public synchronized String getExternalIp() {
             return externalIp;
         }
 
-        if (configFromFiles.hasPath("peer.discovery.external.ip")) {
-            String externalIpFromConfig = configFromFiles.getString("peer.discovery.external.ip").trim();
+        if (configFromFiles.hasPath(PROPERTY_EXTERNAL_IP)) {
+            String externalIpFromConfig = configFromFiles.getString(PROPERTY_EXTERNAL_IP).trim();
             if (!externalIpFromConfig.isEmpty()){
                 try {
-                    InetAddress addr = InetAddress.getByName(externalIpFromConfig);
-                    externalIp = addr.getHostAddress();
+                    InetAddress address = tryParseIpOrThrow(externalIpFromConfig);
+                    externalIp = address.getHostAddress();
                     logger.info("External address identified {}", externalIp);
                     return externalIp;
                 } catch (IOException e) {
-                    externalIp = null;
-                    logger.warn("Can't resolve external address");
+                } catch (IllegalArgumentException e) {
                 }
+                externalIp = null;
+                logger.warn("Can't resolve external address");
             }
         }
 
         externalIp = getMyPublicIpFromRemoteService();
         return externalIp;
     }
 
-    public String getMyPublicIpFromRemoteService(){
-        logger.info("External IP wasn't set or resolved, using checkip.amazonaws.com to identify it...");
+    private String getMyPublicIpFromRemoteService(){
```

Comment:
I prefer to keep it as String since it would require changes on many classes dependant of getExternalIP() since it wouldn't help much to change only the signature of this one.